### ISSUE TITLE
`out=` variant for `conj` unary operator

### ISSUE THREAD

**Issue by krshrimali** [pytorch/pytorch#69287]
## 🚀 Feature

For consistency reasons, will it be worth having an `out` variant for `conj` operator in PyTorch?

## Motivation/Pitch

This came to notice while trying to implement operators for Sparse CSR layout in PyTorch. Some of the ops not having `out` variants make us do extra "specific" stuff for them, like not using the common macros and writing separate implementations. In general as well, I believe, having an `out` variant for `conj` is good for consistency reasons.

cc: @IvanYashchuk @mruberry

**Comment by nikitaved** [comment]
@krshrimali , maybe it could be of help, but there is a `conj_physical` and an in-place `conj_physical_` methods for eager conjugation.

**Comment by Lezcano** [comment]
Yeah, I don't quite understand what would be the application of having `conj_out`, as `conj` just works at the `Tensor` level, not at the `Storage` level. This would be similar to having a `transpose_out`, which we don't have.

**Comment by anjali411** [comment]
As @nikitaved we do have an out-of-place variant for `torch.conj` called `conj_physical`. There's an in-place, out-of-place, and out= variant for `conj_physical`.

Now as @Lezcano mentioned, we don't usually have `out=` variants for in-place operations, so I think it'd be inconsistent to have one for `conj`.
